src/app/clusters/webrtc-transport-provider-server/tests/TestWebRTCTransportProviderCluster.cpp

```diff
@@ -0,0 +1,626 @@
+
+    WebRTCSessionStruct outSession;
+
+    // Test successful provide offer
+    mockDelegate.SetProvideOfferResult(CHIP_NO_ERROR);
+
+    CHIP_ERROR result = mockDelegate.HandleProvideOffer(args, outSession);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_EQ(outSession.id, 456);
+    EXPECT_EQ(outSession.streamUsage, StreamUsageEnum::kRecording);
+    EXPECT_EQ(mockDelegate.GetLastProvideOfferArgs().sdp, "test_sdp_offer");
+
+    // Test error case
+    mockDelegate.SetProvideOfferResult(CHIP_ERROR_INVALID_ARGUMENT);
+    result = mockDelegate.HandleProvideOffer(args, outSession);
+    EXPECT_EQ(result, CHIP_ERROR_INVALID_ARGUMENT);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleProvideAnswer)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
+
+    // Setup test data
+    uint16_t sessionId        = 789;
+    std::string testSdpAnswer = "test_sdp_answer";
+
+    // Test successful answer handling
+    mockDelegate.SetProvideAnswerResult(CHIP_NO_ERROR);
+
+    CHIP_ERROR result = mockDelegate.HandleProvideAnswer(sessionId, testSdpAnswer);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_EQ(mockDelegate.GetLastSessionId(), 789);
+    EXPECT_EQ(mockDelegate.GetLastSdpAnswer(), testSdpAnswer);
+
+    // Test error case
+    mockDelegate.SetProvideAnswerResult(CHIP_ERROR_INTERNAL);
+    result = mockDelegate.HandleProvideAnswer(sessionId, testSdpAnswer);
+    EXPECT_EQ(result, CHIP_ERROR_INTERNAL);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleProvideICECandidates)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
+
+    // Setup test data
+    uint16_t sessionId = 999;
+    std::vector<ICECandidateStruct> testCandidates;
+
+    ICECandidateStruct candidate1;
+    candidate1.candidate = chip::CharSpan("candidate1", 10);
+    testCandidates.push_back(candidate1);
+
+    ICECandidateStruct candidate2;
+    candidate2.candidate = chip::CharSpan("candidate2", 10);
+    testCandidates.push_back(candidate2);
+
+    // Test successful ICE candidates handling
+    mockDelegate.SetProvideICECandidatesResult(CHIP_NO_ERROR);
+
+    CHIP_ERROR result = mockDelegate.HandleProvideICECandidates(sessionId, testCandidates);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_EQ(mockDelegate.GetLastSessionId(), 999);
+    EXPECT_EQ(mockDelegate.GetLastCandidates().size(), 2u);
+
+    // Test error case
+    mockDelegate.SetProvideICECandidatesResult(CHIP_ERROR_INVALID_ARGUMENT);
+    result = mockDelegate.HandleProvideICECandidates(sessionId, testCandidates);
+    EXPECT_EQ(result, CHIP_ERROR_INVALID_ARGUMENT);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleEndSession)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
```

Comment:
all the tests below are accesing mockDelegate. What does the server have to do with anything then? Should we test server instead of the mock delegate?

Applies to tests above as well.